### PR TITLE
feat: Allow table search to include label and value for tables with select column type

### LINKED ISSUE #35406: [Bug]: datatable search doesn't apply to label field of select widget columns
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I have a datatable with a select widget column using value/label pairs then if I use the search box of the datatable the search is done on the value field and not the label field of the select widget column. 

For exemple let say I have a color column with the following values for this column : [{value:1,label:"Red"},{value:2,label:"Blue"}]
If I put "Red" in the datatable search field no rows is displayed, while I would expect to see all the rows with value Red in it. If I put "1" in the search field then "Red" rows will be displayed

### Steps To Reproduce

1. Create a datatable with a select widget column "color". Configure it with select options: [{value:1,label:"Red"},{value:2,label:"Blue"}]
2. Put the following table datas: [{color:1},{color:2}]
3. Enter in the datatable search field "Red"
4. You can see that no rows is displayed when we would expect to see one row that has value Red

### Public Sample App

https://app.appsmith.com/app/datatable-search-doesn-t-apply-to-label-field-of-select-widget-columns/page1-66b08e33f69b2c7c16e2b93b

### Environment

Production

### Severity

Medium (Frustrating UX)

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
**Problem**
The search functionality in the table widget does not properly search through the label values in columns with the select type. Currently, it only searches using the underlying value property, leading to inconsistent and inaccurate search results.

**Root Cause**
A recent [PR](https://github.com/appsmithorg/appsmith/pull/35124) updated the table widget to display label values instead of the raw values in select columns. However, this change inadvertently caused the search logic to overlook these label values, continuing to search only by the value property.

**Solution**
To address this, the search functionality in the derived.js file within tablev2 has been updated. The solution involves retrieving the label value associated with the select column's value property and passing this label into the search logic. This ensures that the search functionality accurately reflects what is displayed in the table, providing correct search results based on the label values.

Fixes #35406

## Automation

/ok-to-test tags="@tag.Sanity, @tag.Widget, @tag.Table, @tag.Binding"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10817317106>
> Commit: f921efc65232c164d0249904ec1ae66dfbb39eae
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10817317106&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity, @tag.Widget, @tag.Table, @tag.Binding`
> Spec:
> <hr>Wed, 11 Sep 2024 19:44:20 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced search functionality in the TableWidgetV2 to include label values for select column types.
	- Improved test specifications for the TableV2 component, incorporating feature flags and complex data interactions.
	- Updated test cases to clarify input formats and validate rendering based on new structures.

- **Bug Fixes**
	- Added test cases to verify the correct behavior of the table filtering and searching mechanisms.

- **Chores**
	- Streamlined input data structure in Cypress tests for better compatibility with the component's configuration requirements.

### CODE REVIEW

**Comment by sagar-qa007** [comment]
@jacquesikot Please check failure.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts [hunk 1]
```diff
@@ -42,7 +42,8 @@ describe(
       cy.updateCodeInput(
         ".t--property-control-options",
         `
-      [
+      {{
```
Comment (rahulbarwal) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/columnTypes/Select1_spec.ts:45]: How was this spec working before?

**Review by rahulbarwal — APPROVED** [review]
LGTM. Just ensure that with feature flag set to true the select label(instead of value) is shown in the table.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/TableV2Filter1_1_Spec.ts [hunk 2]
```diff
@@ -8,6 +9,56 @@ import {
   draggableWidgets,
   agHelper,
 } from "../../../../../support/Objects/ObjectsCore";
+import EditorNavigation, {
+  EntityType,
+} from "../../../../../support/Pages/EditorNavigation";
+
+const demoTableData = `
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/TableV2Filter1_1_Spec.ts:16]: Please move this data in fixture.

Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/TableV2Filter1_1_Spec.ts:16]: Please move this data in fixture for cypress

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

**Review by rahulbarwal — APPROVED** [review]
Please update the description to explain the 3 types of values that can come on select options and how we are handling them. 

* Also findout from Rishabh or Pawan on if there is a way to reuse code in `derived.js` we are repeating our selves which is a code smell.

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**